runtime/src/accounts_db.rs

```diff
@@ -841,6 +842,25 @@ impl AccountsDB {
         infos
     }
 
+    pub fn verify_zero_lamports(&self, _slot: Slot, ancestors: &HashMap<Slot, usize>) -> bool {
+        let accounts_index = self.accounts_index.read().unwrap();
+        !self.scan_accounts(
+            &ancestors,
+            |collector: &mut bool, option| {
+                if let Some((pubkey, account, _b)) = option {
+                    if account.lamports == 0 {
+                        let list = accounts_index.account_maps.get(&pubkey).unwrap().read().unwrap();
+                        let list: Vec<(Slot, AccountInfo)> = list.iter().filter(|(slot, _)| accounts_index.is_root(*slot)).cloned().collect();
+                        error!("ryoqun had zero?: {}, {:?}, {}, list: {}", pubkey, account, _b, list.len());
+                        if list.len() > 1 {
```

Comment:
By this way, I think we can prevent attackers from shielding legitimate accounts as zero balance.